src/components v2/Atoms/Forms/InputText/InputText.vue

```diff
@@ -0,0 +1,169 @@
+<template>
+  <div class="ph-input-text" :class="inFocus && 'ph-input-text-focus'">
+    <div v-if="iconLeft" class="ph-input-text-icon ph-input-text-icon-left">
+      <p-icon :name="iconLeft" type="lg" />
+    </div>
+    <input
+      :id="id"
+      class="ph-input-text-input"
+      :class="{
+        'ph-input-text-input-icon-left': iconLeft,
+        'ph-input-text-icon-right': iconRight,
+        'ph-input-text-icon-center': centered,
+        'input-error': error,
+      }"
+      :placeholder="placeholder"
+      :type="!$attrs.type ? 'text' : number === true ? 'number' : $attrs.type"
+      :disabled="$attrs.disabled"
+      tabindex="0"
+      :value="value"
+      @keydown="validatePress"
+      @focus="onFocus"
+      @blur="onBlur"
+      @wheel="$event.preventDefault()"
+      @input="$emit('input', $event.target.value)"
+    />
+    <div v-if="iconRight" class="ph-input-text-icon">
+      <p-icon :name="iconRight" type="lg" />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue';
+import PIcon from '@/components/Icon';
+import { InputTextData, InputValueType } from './types';
+import { isNumber, isAllowedKey } from '@/utils';
+
+export default Vue.extend({
+  name: 'P2InputText',
+  components: {
+    PIcon,
+  },
+  inheritAttrs: true,
+
+  props: {
+    centered: {
+      type: Boolean as PropType<boolean>,
+      default: false,
+    },
+    number: {
+      type: Boolean as PropType<boolean>,
+      default: false,
+    },
+    placeholder: {
+      type: String as PropType<string>,
+      default: '',
+    },
+    iconLeft: {
+      type: String as PropType<string>,
+      default: '',
+    },
+    iconRight: {
+      type: String as PropType<string>,
+      default: '',
+    },
+    error: {
+      type: String as PropType<string>,
+      default: '',
+    },
+    value: {
+      type: [Number, String] as PropType<InputValueType>,
+      default: null,
+    },
+  },
+
+  data(): InputTextData {
+    return {
+      inFocus: false,
+      currencyOptions: {
+        currency: null,
+        precision: 0,
+        valueRange: { min: 0 },
+        allowNegative: false,
+      },
+      id: '',
+    } as InputTextData;
+  },
+
+  mounted() {
+    this.id = 'photon_input_' + this._uid;
+  },
```

Comment:
**discussion:** Allow parent to handle full id.

Let's remove the prefix here and allow the users of this component more freedom over the component. If the user wants to make it a `photon_input_` with the id then we can just let them pass that in.